### PR TITLE
PLANNER-865: Workbench: ScoreHolder support in GuidedDecisionTables causes error in server log

### PR DESCRIPTION
**Pull request by mcimbora**
Part of:
- https://github.com/kiegroup/drools/pull/1476
- https://github.com/kiegroup/drools-wb/pull/620
- https://github.com/kiegroup/optaplanner-wb/pull/222
- https://github.com/kiegroup/droolsjbpm-integration/pull/1193

### CODE REVIEW

File: drools-workbench-models/drools-workbench-models-guided-dtable/src/main/java/org/drools/workbench/models/guided/dtable/backend/GuidedDTDRLPersistence.java [hunk 1]
```diff
@@ -92,12 +92,6 @@ public static GuidedDTDRLPersistence getInstance() {
     }
 
     public String marshal(final GuidedDecisionTable52 originalDTable) {
-        return marshal(originalDTable,
-                       Collections.emptyList());
-    }
-
-    public String marshal(final GuidedDecisionTable52 originalDTable,
-                          final Collection<RuleModelIActionPersistenceExtension> extensions) {
```
Comment (jomarko) [inline drools-workbench-models/drools-workbench-models-guided-dtable/src/main/java/org/drools/workbench/models/guided/dtable/backend/GuidedDTDRLPersistence.java:100]: Unused imports

File: drools-workbench-models/drools-workbench-models-guided-dtable/src/main/java/org/drools/workbench/models/guided/dtable/backend/util/GuidedDTBRDRLPersistence.java [hunk 2]
```diff
@@ -79,15 +79,13 @@ protected LHSPatternVisitor getLHSPatternVisitor(final boolean isDSLEnhanced,
     protected RHSActionVisitor getRHSActionVisitor(final boolean isDSLEnhanced,
                                                    final StringBuilder buf,
                                                    final String indentation,
-                                                   final RHSGeneratorContextFactory generatorContextFactory,
-                                                   final Collection<RuleModelIActionPersistenceExtension> extenstions) {
```
Comment (jomarko) [inline drools-workbench-models/drools-workbench-models-guided-dtable/src/main/java/org/drools/workbench/models/guided/dtable/backend/util/GuidedDTBRDRLPersistence.java:83]: unused imports

File: drools-workbench-models/drools-workbench-models-guided-template/src/main/java/org/drools/workbench/models/guided/template/backend/RuleTemplateModelDRLPersistenceImpl.java [hunk 3]
```diff
@@ -98,14 +98,12 @@ protected LHSPatternVisitor getLHSPatternVisitor(final boolean isDSLEnhanced,
     protected RHSActionVisitor getRHSActionVisitor(final boolean isDSLEnhanced,
                                                    final StringBuilder buf,
                                                    final String indentation,
-                                                   final RHSGeneratorContextFactory generatorContextFactory,
-                                                   final Collection<RuleModelIActionPersistenceExtension> extensions) {
```
Comment (jomarko) [inline drools-workbench-models/drools-workbench-models-guided-template/src/main/java/org/drools/workbench/models/guided/template/backend/RuleTemplateModelDRLPersistenceImpl.java:102]: unused imports

**Review by jomarko — COMMENTED** [review]
@mcimbora Just unused imports, I had look on Model Persistence implementations, Thanks to inheritance it seems that coverage is enough.

File: drools-workbench-models/drools-workbench-models-commons/src/main/java/org/drools/workbench/models/commons/backend/rule/RuleModelDRLPersistenceImpl.java [hunk 4]
```diff
@@ -2829,7 +2811,9 @@ private void parseRhs(final RuleModel m,
                     } else if (matchingExtensionsDslLine.size() > 1) {
                         throw new RuleModelDRLPersistenceException("Ambiguous RuleModelIActionPersistenceExtension implementations (" + matchingExtensionsDslLine + ") found for line " + line);
                     } else {
-                        m.addRhsItem(matchingExtensionsDslLine.get(0).unmarshal(dslLine));
+                        unmarshalUsingExtension(m,
```
Comment (manstis) [inline drools-workbench-models/drools-workbench-models-commons/src/main/java/org/drools/workbench/models/commons/backend/rule/RuleModelDRLPersistenceImpl.java:2814]: Formatting

File: drools-workbench-models/drools-workbench-models-commons/src/test/java/org/drools/workbench/models/commons/backend/rule/PluggableIActionTest.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+    @Test
+    public void marshalPluggableIAction() {
+        RuleModel ruleModel = new RuleModel();
+        ruleModel.name = "TestIAction rule";
+        ruleModel.addRhsItem(new TestIAction());
+
+        String marshaledString = RuleModelDRLPersistenceImpl.getInstance().marshal(ruleModel);
+
+        assertEquals(DRL_RULE,
```
Comment (manstis) [inline drools-workbench-models/drools-workbench-models-commons/src/test/java/org/drools/workbench/models/commons/backend/rule/PluggableIActionTest.java:43]: Formatting

**Review by manstis — COMMENTED** [review]
A couple of petty minor formatting issues to fix along with the imports.. otherwise looks OK to me.

**Review by jomarko — APPROVED** [review]

**Review by manstis — APPROVED** [review]

**Pull request merged**